src/data/fin/tuple/basic.lean

```diff
@@ -120,19 +120,58 @@ end
 
 /-- Recurse on an `n+1`-tuple by splitting it into a single element and an `n`-tuple. -/
 @[elab_as_eliminator]
-def cons_induction {P : (Π i : fin n.succ, α i) → Sort v}
+def cons_cases {P : (Π i : fin n.succ, α i) → Sort v}
   (h : ∀ x₀ x, P (fin.cons x₀ x)) (x : (Π i : fin n.succ, α i)) : P x :=
 _root_.cast (by rw cons_self_tail) $ h (x 0) (tail x)
 
-@[simp] lemma cons_induction_cons {P : (Π i : fin n.succ, α i) → Sort v}
+@[simp] lemma cons_cases_cons {P : (Π i : fin n.succ, α i) → Sort v}
   (h : Π x₀ x, P (fin.cons x₀ x)) (x₀ : α 0) (x : Π i : fin n, α i.succ) :
-  @cons_induction _ _ _ h (cons x₀ x) = h x₀ x :=
+  @cons_cases _ _ _ h (cons x₀ x) = h x₀ x :=
 begin
-  rw [cons_induction, cast_eq],
+  rw [cons_cases, cast_eq],
   congr',
   exact tail_cons _ _
 end
 
+/-- Recurse on an tuple by splitting into `fin.elim0` and `fin.cons`. -/
+@[elab_as_eliminator]
+def cons_induction {α : Type*} {P : Π {n : ℕ}, (fin n → α) → Sort v}
+  (h0 : P fin.elim0)
+  (h : ∀ {n} x₀ (x : fin n → α), P x → P (fin.cons x₀ x)) : Π {n : ℕ} (x : fin n → α), P x
+| 0 x := by convert h0
+| (n + 1) x := cons_cases (λ x₀ x, h _ _ $ cons_induction _) x
+
+lemma cons_injective_of_injective {α} {x₀ : α} {x : fin n → α} (hx₀ : x₀ ∉ set.range x)
```

Comment:
I don't mind this name so much, this suggestion is more to match with the `nodup_of_fn` suggestion:
```suggestion
lemma _root_.function.injective.fin_cons {α} {x₀ : α} {x : fin n → α} (hx₀ : x₀ ∉ set.range x)
```